### DISCUSSION TITLE
pnpm tauri ios dev failing?

### DISCUSSION THREAD

**Discussion by scullionw** [tauri-apps/tauri#6382] [Q&A]
trying out the new mobile dev detup with create-tauri-app 3.0, but getting `ld: library not found for -ltauri_app` when running ` pnpm tauri ios dev`?

**Comment by FabianLars** [comment]
This might be an issue with the cta template and/or with how the library name is resolved (because cta overwrites the default lib name)

cc @amrbashir

**Comment by amrbashir** [comment]
Looks like I forgot to update the xcode project file upstream in `tauri-mobile` and in `tauri-cli`, until a new version is released please open `project.yml` and replace each occurrence of `libtauri-app.a` with `lib<lib_name_in_Cargo.toml>.a` in your case it should be `libtauri_applib.a`

EDIT: PR is up https://github.com/tauri-apps/tauri/pull/6387 but unfortunately, I don't have a macOS to test, could you see if it fixes the issue for you? remember to remove anything related to the ios/xcode project before testing it out
```bash
cargo install tauri-cli --git https://github.com/tauri-apps --branch fix/lib_name-xcode-script
cargo tauri ios init
cargo tauri ios dev
```

**Comment by scullionw** [comment]
ok i am now getting a different error:
<img width="1723" alt="Screenshot 2023-03-02 at 9 54 39 PM" src="https://user-images.githubusercontent.com/11668344/222550000-dbfa2c05-2c99-45ad-be2b-76555df2b4db.png">

**Comment by joeyfigaro** [comment]
This is still occurring, along with @scullionw's last reported `command ["xcodebuild"] exited with code 65` error. 

Output of `pnpm tauri info`:
```sh
[✔] Environment
    - OS: Mac OS 14.1.1 X64
    ✔ Xcode Command Line Tools: installed
    ✔ rustc: 1.75.0 (82e1608df 2023-12-21)
    ✔ cargo: 1.75.0 (1d8b05cdd 2023-11-20)
    ✔ rustup: 1.26.0 (5af9b9484 2023-04-05)
    ✔ Rust toolchain: stable-aarch64-apple-darwin (default)
    - node: 20.10.0
    - pnpm: 8.15.4
    - yarn: 1.22.21
    - npm: 10.2.3
    - bun: 1.0.35

[-] Packages
    - tauri [RUST]: 2.0.0-beta.13
    - tauri-build [RUST]: 2.0.0-beta.10
    - wry [RUST]: 0.37.0
    - tao [RUST]: 0.26.1
    - @tauri-apps/api [NPM]: 2.0.0-beta.6
    - @tauri-apps/cli [NPM]: 2.0.0-beta.11

[-] App
    - build-type: bundle
    - CSP: unset
    - frontendDist: ../dist
    - devUrl: http://localhost:1420/
    - framework: React
    - bundler: Vite
```

and the missing pnpm error from my latest iOS build:
```sh
/Users/joeyfigaro/Library/Developer/Xcode/DerivedData/tonepad-clxazsiahclvakfscznyxutulzoq/Build/Intermediates.noindex/tonepad.build/debug-iphonesimulator/tonepad_iOS.build/Script-0C56976A17D438BD5C1334CE.sh: line 2: pnpm: command not found
Command PhaseScriptExecution failed with a nonzero exit code
```

**Comment by jqpe** [comment]
For some reason Xcode will not read PATH (at least the current user's) and defaults to looking under /usr/local/bin/*. Launching a simulator with `tauri ios dev 0 --open` also **DOES NOT** work, but you can open the simulator from Xcode. This is an issue with Xcode and not so much Tauri.

For me linking the pnpm and node executables to /usr/local/bin resolved it:

```sh
# for pnpm
ln -s $(which pnpm) /usr/local/bin/pnpm
# for node
ln -s $(which node) /usr/local/bin/node
```
